### PR TITLE
Add 'take' and 'skip' functions to reader

### LINKED ISSUE #34: Feature request: offset and limit
It would be nice if you could specify an `offset` and `limit` when reading a file, in order to make pagination possible/easier.

### PR DESCRIPTION
**Pull request by ahallhognason**
Closes #34 

Instead of `offset` and `limit`, I used `take` and `skip` to follow the conventions from LazyCollections.

### CODE REVIEW

File: src/SimpleExcelReader.php [hunk 1]
```diff
@@ -18,6 +18,12 @@ class SimpleExcelReader
 
     private bool $processHeader = true;
 
+    private $skip = 0;
```
Comment (freekmurze) [inline src/SimpleExcelReader.php:21]: Typehint all properties

File: src/SimpleExcelReader.php [hunk 2]
```diff
@@ -18,6 +18,12 @@ class SimpleExcelReader
...
+
+    private $limit = 0;
+
+    private $use_limit = false;
```
Comment (freekmurze) [inline src/SimpleExcelReader.php:25]: Use camelCase

File: src/SimpleExcelReader.php [hunk 3]
```diff
@@ -61,6 +67,21 @@ public function getReader(): ReaderInterface
...
     }
 
+    public function skip($count)
+    {
+        $this->skip = $count;
+
+        return $this;
+    }
+
+    public function take($count)
```
Comment (freekmurze) [inline src/SimpleExcelReader.php:77]: Add type hint and return type hint

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you add an example to the readme?

**Review by ahallhognason — COMMENTED** [review]
Made all suggested changes.

**Comment by freekmurze** [comment]
Thanks!

**Comment by thijsw** [comment]
@ahallhognason Thank you for this PR, much welcomed!

**Comment by Francismori7** [comment]
This PR broke `getRows()`, if called multiple times one after the other on the same file instance. See #36

**Comment by freekmurze** [comment]
@Francismori7 Could you send a PR for this fix. Also maybe add a test, so we don't encounter this regression again.

**Pull request merged**